### PR TITLE
UtilsLike type does not support router sub-types

### LINKED ISSUE #4140: bug: router-factory/polymorphism UtilsLike type does not support router sub-types via mergeRouters
### Provide environment information

See PR in our own codebase/environment

### Describe the bug

We can create a router factory like the below, and generate abstract types from it using our polymorphism types.

```ts
function createMyRouter() {
  return t.router({ foo, bar })
}

type MyRouter = ReturnType<typeof createMyRouter>;

export type MyRouterRouteLike = RouterLike<MyRouter>;

export type MyRouterUtilsLike = UtilsLike<MyRouter>;
```

If we then use it like 

```ts
const someRouter = t.mergeRouters(
  createMyRouter(),
  t.router({
    extraProc: t.procedure
  })
)
```

Now when we try to pass this router's utils to the `UtilsLike` result, it gets rejected as incompatible by TypeScript, similar to below

image

### Link to reproduction

-

### To reproduce

Reproduction in PR test

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### PR DESCRIPTION
**Pull request by Nick-Lucas**
Closes #4140 

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/react-query/src/shared/proxy/utilsProxy.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ type DecorateProcedure<
       predicate?: (
         query: Query<
           inferProcedureInput<TProcedure>,
-          TRPCClientError<TRouter>,
+          TRPCClientError<TProcedure>,
```
Comment (Nick-Lucas) [inline packages/react-query/src/shared/proxy/utilsProxy.ts:110]: So this was encoding whole router information into the type defs for a single procedure, which broke this one polymorphism corner case. 

Not sure what impacts this will really have though

File: packages/tests/server/react/polymorphism.test.tsx [hunk 2]
```diff
@@ -186,6 +199,63 @@ describe('polymorphism', () => {
...
+      function DiscussionsExportPage() {
+        const utils = trpc.useContext();
+
+        const [currentExport, setCurrentExport] = useState<null | number>(null);
+
+        return (
+          <>
+            <StartExportButton
+              route={trpc.github.discussions.export}
+              utils={utils.github.discussions.export}
```
Comment (Nick-Lucas) [inline packages/tests/server/react/polymorphism.test.tsx:215]: This line gave a typescript error

**Comment by Nick-Lucas** [comment]
This commit created the issue: https://github.com/trpc/trpc/commit/b05d207df136303bcbae47fbab55a321fc5dcd15

File: packages/react-query/src/shared/proxy/utilsProxy.ts (hunk 1 above)
Comment (KATT) [reply packages/react-query/src/shared/proxy/utilsProxy.ts:110]: `TRPCClientError<TProcedure>` is what we should be using everywhere 👌

**Review by KATT — DISMISSED** [review]

File: packages/react-query/src/shared/proxy/utilsProxy.ts (hunk 1 above)
Comment (Nick-Lucas) [reply packages/react-query/src/shared/proxy/utilsProxy.ts:107]: Computer says 🤔

image

**Review by sachinraja — APPROVED** [review]
